src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -138,11 +139,32 @@ private async Task<IAccount> GetSingleAccountForLoginHintAsync(string loginHint)
         {
             if (!string.IsNullOrEmpty(loginHint))
             {
-                IReadOnlyList<IAccount> accounts = (await CacheManager.GetAccountsAsync().ConfigureAwait(false))
+                IList<IAccount> accounts = (await CacheManager.GetAccountsAsync().ConfigureAwait(false))
                     .Where(a => !string.IsNullOrWhiteSpace(a.Username) &&
                            a.Username.Equals(loginHint, StringComparison.OrdinalIgnoreCase))
                     .ToList();
 
+                if (accounts.Count == 0 && DesktopOsHelper.IsWindows())
+                {
+                    string currentUpn;
+                    
+                    try
+                    {
+                        currentUpn = await ServiceBundle.PlatformProxy.GetUserPrincipalNameAsync();
+                    }
+                    catch
+                    {
+                        // NetDesktopPlatformProxy throws instead of returning string.Empty.
+                        currentUpn = null;
+                    }
+
+
+                    if (currentUpn == loginHint)

```

Comment:
also if `string.IsNotNullOrEmpty(currentUpn)`, otherwise this might login the current OS account on .NET Classic.

Reply:
This shouldn't happen as this is all under `if (!string.IsNullOrEmpty(loginHint))` but it doesn't hurt to make the code clearer :)